src/main/java/frc/robot/subsystems/Vision/VisionIOReal.java

```diff
@@ -4,21 +4,19 @@
 
 package frc.robot.subsystems.Vision;
 
-import org.photonvision.PhotonCamera;
-
 import edu.wpi.first.math.geometry.Pose3d;
+import org.photonvision.PhotonCamera;
 
 /** Add your docs here. */
 public class VisionIOReal implements VisionIO {
-    PhotonCamera camera = new PhotonCamera("opi camera");
-
-    @Override
-    public void updateInputs(VisionIOInputs inputs, Pose3d robotPose) {
-        var latestResult = camera.getLatestResult();
+  PhotonCamera camera = new PhotonCamera("opi camera");
 
-        inputs.timestamp = latestResult.getTimestampSeconds();
-        inputs.timeSinceLastTimestamp = latestResult.getLatencyMillis();
-        inputs.targets = latestResult.targets;
-    }
+  @Override
+  public void updateInputs(VisionIOInputs inputs, Pose3d robotPose) {
+    var latestResult = camera.getLatestResult();
 
+    inputs.timestamp = latestResult.getTimestampSeconds();
+    inputs.timeSinceLastTimestamp = latestResult.getLatencyMillis();
+    inputs.targets = latestResult.targets;
```

Comment:
make sure to assign a value to number of tags

Reply:
numtags isn't passed in tho? its just assigned in toLog, unless i'm misunderstanding something

Reply:
ah right.